### PR TITLE
Fix TRD tracking in standalone version for TFs

### PR DESCRIPTION
**Pull request by martenole**
Before the tracker could not handle a TF with multiple collisions. I did not notice it so far, since I only tested with the box generator and a single event..
@davidrohr for the GPU track type I need a time estimate, similar to the `TrackTPC::getTime0()`. Directly in `GPUTPCGMMergedTrack` I could not see it. I assume I need to translate the track's Z into a time coordinate (using a vertex estimate and track eta?) in order to find the matching TRD trigger times. Can I do this using the default TPC vDrift value? And then open the road rather wide in Z accordingly?

There are some unrelated minor code changes in some of the TRD libraries in this PR, since as I learned today we might enable some additional compiler warnings in the future (`-Wreorder` for example).

### CODE REVIEW

File: GPU/GPUTracking/Global/GPUChainTrackingTRD.cxx [hunk 1]
```diff
@@ -37,6 +37,7 @@ int GPUChainTracking::RunTRDTracking()
   if (mIOPtrs.nTRDTracklets == 0) {
     return 0;
   }
+  Tracker.SetProcessPerTimeFrame(true);
```
Comment (davidrohr) [inline GPU/GPUTracking/Global/GPUChainTrackingTRD.cxx]: Do we want to enable this unconditionally? I mean, there is still the possibility that we process triggered events.
I think you should check that `GPUParam::par.continuousMaxTimeBin != 0` for time frames, = 0 means triggered event.

File: GPU/GPUTracking/Global/GPUChainTrackingTRD.cxx [hunk 2]
```diff
@@ -52,7 +53,7 @@ int GPUChainTracking::RunTRDTracking()
       if (!Tracker.CheckTrackTRDCandidate(trktrd)) {
         continue;
       }
-
+      GPUWarning("It does not make sense to load input track without time window information, except when processing Run 2 data");
```
Comment (davidrohr) [inline GPU/GPUTracking/Global/GPUChainTrackingTRD.cxx]: Also this should depend on whether we process a time frame or not.
The conversion to O2 tracks gets the time0 as:
```
time0 = tracks[i].GetParam().GetTZOffset();
```
The problem is that the deltaTFwd and deltaTBwd is not so obvious.
For this the tracker iterates over the TPC clusters. Perhaps for now you can just set 1 TPC drift time? That should at least work. And this mode of operation is anyway mostly for tests, the default will be using the o2 track I guess.

**Review by davidrohr — CHANGES_REQUESTED** [review]

File: GPU/GPUTracking/Global/GPUChainTrackingTRD.cxx (hunk 1 above)
Comment (martenole) [reply GPU/GPUTracking/Global/GPUChainTrackingTRD.cxx]: Thanks I did not know that this option already existed. So I am using it now

File: GPU/GPUTracking/Global/GPUChainTrackingTRD.cxx (hunk 2 above)
Comment (martenole) [reply GPU/GPUTracking/Global/GPUChainTrackingTRD.cxx]: Does `GetTZOffset()` have a similar precision as the time of the `TrackTPC`? At least when running with a single thread and looking at the times of the GPU and O2 track types they did not really match (for the same track index). And then in O2 I check if the track is on the A/C side or if it crosses the CE. In the latter case the time should be precise and I check only one TRD trigger time. Trying to do the same now with the GPU track types I find much less tracks than with the O2 track version (roughly factor 10)

Comment (martenole) [reply GPU/GPUTracking/Global/GPUChainTrackingTRD.cxx]: I had the wrong unit for the TPC drift time... This works now as expected

**Comment by martenole** [comment]
ping @davidrohr

**Pull request merged**